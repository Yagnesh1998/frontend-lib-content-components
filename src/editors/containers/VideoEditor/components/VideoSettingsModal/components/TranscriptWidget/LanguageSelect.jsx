import React from 'react';
import PropTypes from 'prop-types';

import {
  Form,
} from '@edx/paragon';
import { connect, useDispatch } from 'react-redux';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import hooks from './hooks';
import { selectors } from '../../../../../../data/redux';
import { videoTranscriptLanguages } from '../../../../../../data/constants/video';
import messages from './messages';

export const LanguageSelect = ({
  title, // For a unique id for the form control
  language,
  // Redux
  openLanguages, // Only allow those languages not already associated with a transcript to be selected
  transcripts,
  // intl
  intl,

}) => {
  const onLanguageChange = hooks.onSelectLanguage({
    fileName: title, dispatch: useDispatch(), transcripts, languageBeforeChange: language,
  });

  return (
    <Form.Group controlId={`selectLanguage-form-${title}`} className="mt-2 mx-2">
      <Form.Control as="select" defaultValue={language} onChange={(e) => onLanguageChange(e)} floatingLabel={intl.formatMessage(messages.languageSelectLabel)}>
        <option value={language}>{videoTranscriptLanguages[language]}</option>
        {Object.entries(videoTranscriptLanguages).map(([lang, text]) => {
          if (language === lang) { return (<></>); }
          if (openLanguages.some(row => row.includes(lang))) {
            return (<option value={lang}>{text}</option>);
          }
          return (<option value={lang} disabled>{text}</option>);
        })}

      </Form.Control>
    </Form.Group>
  );
};

LanguageSelect.defaultProps = {
  openLanguages: [],
};

LanguageSelect.propTypes = {
  openLanguages: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  transcripts: PropTypes.objectOf(PropTypes.string).isRequired,
  intl: intlShape.isRequired,
};

export const mapStateToProps = (state) => ({
  openLanguages: selectors.video.openLanguages(state),
  transcripts: selectors.video.transcripts(state),
});

export const mapDispatchToProps = {};

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(LanguageSelect));
