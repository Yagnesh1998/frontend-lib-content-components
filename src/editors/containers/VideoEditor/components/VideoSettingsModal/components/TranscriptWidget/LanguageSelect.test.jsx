import React from 'react';
import { shallow } from 'enzyme';
import { LanguageSelect } from './LanguageSelect';

const lang1 = 'kLinGon';
const lang1Code = 'kl';
const lang2 = 'eLvIsh';
const lang2Code = 'el';
const lang3 = 'sImLisH';
const lang3Code = 'sl';

jest.mock('../../../../../../data/constants/video', () => ({
  videoTranscriptLanguages: {
    [lang1Code]: lang1,
    [lang2Code]: lang2,
    [lang3Code]: lang3,
  },
}));

describe('LanguageSelect', () => {
  const props = {
    onSelect: jest.fn().mockName('props.OnSelect'),
    title: 'tITle',
    language: lang1Code,
    openLanguages: [[lang2Code, lang2], [lang3Code, lang3]],

  };
  describe('snapshot', () => {
    test('transcript option', () => {
      expect(
        shallow(<LanguageSelect {...props} />),
      ).toMatchSnapshot();
    });
  });
  describe('snapshots -- no', () => {
    test('transcripts no Open Languages, all should be disabled', () => {
      expect(
        shallow(<LanguageSelect {...props} openLanguages={[]} />),
      ).toMatchSnapshot();
    });
  });
});
