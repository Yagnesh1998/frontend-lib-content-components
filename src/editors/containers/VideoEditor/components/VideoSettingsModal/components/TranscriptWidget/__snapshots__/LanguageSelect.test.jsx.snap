// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelect snapshot transcript option 1`] = `
<Form.Group
  className="mt-2 mx-2"
  controlId="selectLanguage-form-tITle"
>
  <Form.Control
    as="select"
    defaultValue="kl"
    floatingLabel="Language"
    onChange={[Function]}
  >
    <option
      value="kl"
    >
      kLinGon
    </option>
    <option
      value="el"
    >
      eLvIsh
    </option>
    <option
      value="sl"
    >
      sImLisH
    </option>
  </Form.Control>
</Form.Group>
`;

exports[`LanguageSelect snapshots -- no transcripts no Open Languages, all should be disabled 1`] = `
<Form.Group
  className="mt-2 mx-2"
  controlId="selectLanguage-form-tITle"
>
  <Form.Control
    as="select"
    defaultValue="kl"
    floatingLabel="Language"
    onChange={[Function]}
  >
    <option
      value="kl"
    >
      kLinGon
    </option>
    <option
      disabled={true}
      value="el"
    >
      eLvIsh
    </option>
    <option
      disabled={true}
      value="sl"
    >
      sImLisH
    </option>
  </Form.Control>
</Form.Group>
`;
